api/v1alpha1/clusterinstance_types.go

```diff
@@ -215,6 +215,31 @@ const (
 	ClusterTypeHighlyAvailable ClusterType = "HighlyAvailable"
 )
 
+type preservationMode string
+
+const (
+	None            preservationMode = "None"
+	All             preservationMode = "All"
+	ClusterIdentity preservationMode = "ClusterIdentity"
+)
+
+type ReinstallSpec struct {
+	// Desired reinstall generation. Used to trigger a reinstallation.
+	// +required
+	Generation string `json:"generation"`
+	// This field informs the SiteConfig Operator whether to preserve data or not.
+	// Furthermore, it provides a mechanism to ensure / enforce that cluster identity data is preserved if
+	// configured to ClusterIdentity.
+	// Possible values:
+	// - None: no data preservation is required.
+	// - All: All Secrets and / or ConfigMaps with the preservation-label are preserved.
+	// - ClusterIdentity: only Secrets and / or ConfigMaps with the cluster-identity preservation-label are preserved.
```

Comment:
Can this be more specific about the label key and value?

Reply:
Updated